src/Service/FileUploader.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Service;
+
+use Symfony\Component\HttpFoundation\File\Exception\FileException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\String\Slugger\SluggerInterface;
+
+class FileUploader
+{
+    private $targetDirectory;
+    private $slugger;
+
+    public function __construct($targetDirectory, SluggerInterface $slugger)
+    {
+        $this->targetDirectory = $targetDirectory;
+        $this->slugger = $slugger;
+    }
+
+    public function upload(UploadedFile $file)
+    {
+        $originalFilename = pathinfo($file->getClientOriginalName(), PATHINFO_FILENAME);
+        $safeFilename = $this->slugger->slug($originalFilename);
+        $fileName = $safeFilename.'-'.uniqid().'.'.$file->guessExtension();
+
+        try {
+            $file->move($this->getTargetDirectory(), $fileName);
+        } catch (FileException $e) {
+
+            return $e->getMessage();
+            //todo : handle exception and display message in front
+        }
+
+        return '/img/tricks/'.$fileName;
```

Comment:
meh. I understand why, but then if you change the parameter in the config, it won't work here anymore

Reply:
fixed on media-video branch